cpp/src/phonenumbers/stringutil.cc

```diff
@@ -143,47 +117,20 @@ void strrmm(string* s, const string& chars) {
 int GlobalReplaceSubstring(const string& substring,
                            const string& replacement,
                            string* s) {
-  assert(s != NULL);
-  if (s->empty() || substring.empty())
-    return 0;
-  string tmp;
-  int num_replacements = 0;
-  int pos = 0;
-  for (size_t match_pos = s->find(substring.data(), pos, substring.length());
-       match_pos != string::npos;
-       pos = static_cast<int>(match_pos + substring.length()),
-          match_pos = s->find(substring.data(), pos, substring.length())) {
-    ++num_replacements;
-    // Append the original content before the match.
-    tmp.append(*s, pos, match_pos - pos);
-    // Append the replacement for the match.
-    tmp.append(replacement.begin(), replacement.end());
-  }
-  // Append the content after the last match.
-  tmp.append(*s, pos, s->length() - pos);
-  s->swap(tmp);
-  return num_replacements;
+  return absl::StrReplaceAll({{substring, replacement}}, s);;
 }
 
 // StringHolder class
-
 StringHolder::StringHolder(const string& s)
-  : string_(&s),
-    cstring_(NULL),
-    len_(s.size())
+  : absl::AlphaNum(s)
 {}
 
-StringHolder::StringHolder(const char* s)
-  : string_(NULL),
-    cstring_(s),
-    len_(std::strlen(s))
+StringHolder::StringHolder(const char* cp)
+  : absl::AlphaNum(cp)
 {}
 
 StringHolder::StringHolder(uint64 n)
-  : converted_string_(SimpleItoa(n)),
-    string_(&converted_string_),
-    cstring_(NULL),
-    len_(converted_string_.length())
+  : absl::AlphaNum(n)
```

Comment:
Note, if StringHolder gets a char, it might convert it in an undesired way. Possibly was true already before & maybe that never happens?

Reply:
Right.. Will test it with an UT just I am also curious here :) code: https://github.com/abseil/abseil-cpp/blob/master/absl/strings/str_cat.h#L263
As you mentioned, we do not have such a use case yet; so will take it offline.